### DISCUSSION TITLE
tar_read has problems reading leaflet objects with provider tiles

### DISCUSSION THREAD

**Discussion by michaelgaunt404** [ropensci/targets#937] [Help]
I can make/read leaflet targets but cannot tar_read leaflet objects that use [provider tiles](https://rstudio.github.io/leaflet/basemaps.html).

Error:
![image](https://user-images.githubusercontent.com/60335544/192629041-515fe5b7-ca9e-4337-82df-8238ddcad32b.png)

Example code - first function works and loads, the second does not:

```
test_leaflet = function(){
  sd = quakes[sample(nrow(quakes), 100),]
  leaflet(sd) %>% addTiles() %>% addMarkers()
}

test_leaflet_providers = function(){
  sd = quakes[sample(nrow(quakes), 100),]
  leaflet(sd) %>% 
    addProviderTiles(providers$Stamen.Toner, group = "Toner") %>%
    addProviderTiles(providers$Stamen.TonerLite, group = "Toner Lite") %>%
    addLayersControl(
      baseGroups = c("OSM (default)", "Toner", "Toner Lite"),
      overlayGroups = c("Mile Posts"),
      options = layersControlOptions(collapsed = FALSE)
    ) %>% addMarkers()
}
```

**Comment by wlandau** [comment]
This is similar to https://github.com/ropensci/targets/discussions/947. The pipeline saves each target in storage (by default, with saveRDS()). Not every R object can be saved this way and then loaded back into the R session: https://cran.r-project.org/web/packages/future/vignettes/future-4-non-exportable-objects.html. For `leaflet` objects, I recommend you save the object in a way that is completely self-contained and readable in a new session even after the temporary JS files disappear. Unfortunately, I do not know enough about `leaflet` to provide a specific recommendation. But to test that it worked, you could save the object with `saveRDS()`, then restart R, and then try to read it again with `readRDS()`.
